packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -402,6 +407,7 @@ class XcodeProjectBuildContext {
     this.environmentType = EnvironmentType.physical,
     this.deviceId,
     this.target,
+    this.isMacOS = false,
```

Comment:
Might make sense to instead change `environmentType` to instead match an enum like `xcodebuild -showsdks`
iOS, iOS simulator, macOS, etc.  I think `EnvironmentType` is only used for iOS right now so you could change that, but I may be wrong.

Reply:
Thanks for the suggestion! I'm much happier with how this looks now :)